File: cloud-manager-web/src/main/resources/static/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../ItemCat/itemCat/findAll');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../ItemCat/itemCat/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../ItemCat/itemCat/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../ItemCat/itemCat/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../ItemCat/itemCat/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../ItemCat/itemCat/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../ItemCat/itemCat/search?page='+page+"&rows="+rows, searchEntity);
	}

	//根据上级ID查询下级列表
	this.findByParentId=function(parentId,page,rows){
		return $http.get('../ItemCat/itemCat/findByParentId?parentId='+parentId+"&page="+page+"&rows="+rows);
	}
	this.getTypeTemplateData=function () {
		return $http.get('../TypeTemplate/typeTemplate/TypeTemplateData');
	}
});

File: cloud-sellergoods-service/src/main/java/com/jessie/mall/sellergoodsService/service/ItemCatService.java
package com.jessie.mall.sellergoodsService.service;

import com.jessie.mall.common.common.pageResult;
import com.jessie.mall.pojo.TbItemCat;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ItemCatService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbItemCat> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public pageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbItemCat itemCat);
	
	
	/**
	 * 修改
	 */
	public void update(TbItemCat itemCat);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbItemCat findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public pageResult findPage(TbItemCat itemCat, int pageNum, int pageSize);

	/**
	 * 根据上级ID返回列表
	 * @return
	 */
	public pageResult findByParentId(Long parentId, int pageNum, int pageSize);


	List<TbItemCat> findByParentId(Long parentId);
}
